pkg/providers/ovirt/validation/vm-import-validation.go

```diff
@@ -0,0 +1,124 @@
+	validators.VMDisplayTypeID:                   log,
+	validators.VMHasIllegalImagesID:              block,
+	validators.VMHighAvailabilityPriorityID:      log,
+	validators.VMIoThreadsID:                     warn,
+	validators.VMMemoryPolicyBallooningID:        log,
+	validators.VMMemoryPolicyOvercommitPercentID: log,
+	validators.VMMemoryPolicyGuaranteedID:        log,
+	validators.VMMigrationID:                     log,
+	validators.VMMigrationDowntimeID:             log,
+	validators.VMNumaTuneModeID:                  warn,
+	validators.VMOriginID:                        block,
+	validators.VMRngDeviceSourceID:               log,
+	validators.VMSoundcardEnabledID:              warn,
+	validators.VMStartPausedID:                   log,
+	validators.VMStorageErrorResumeBehaviourID:   log,
+	validators.VMTunnelMigrationID:               warn,
+	validators.VMUsbID:                           block,
+	validators.VMGraphicConsolesID:               log,
+	validators.VMHostDevicesID:                   log,
+	validators.VMReportedDevicesID:               log,
+	validators.VMQuotaID:                         log,
+	validators.VMWatchdogsID:                     block,
+	validators.VMCdromsID:                        log,
+	validators.VMFloppiesID:                      log,
+}
+
+// Validator validates different properties of a VM
+type Validator interface {
+	ValidateVM(vm *ovirtsdk.Vm) []validators.ValidationFailure
+	ValidateDiskAttachments(diskAttachments []*ovirtsdk.DiskAttachment) []validators.ValidationFailure
+	ValidateNics(nics []*ovirtsdk.Nic) []validators.ValidationFailure
+}
+
+// VirtualMachineImportValidator validates VirtualMachineImport object
+type VirtualMachineImportValidator struct {
+	Validator Validator
+}
+
+// NewVirtualMachineImportValidator creates ready-to-use NewVirtualMachineImportValidator
+func NewVirtualMachineImportValidator() VirtualMachineImportValidator {
+	return VirtualMachineImportValidator{
+		Validator: &validators.ValidatorWrapper{},
+	}
+}
+
+// Validate validates whether VM described in VirtualMachineImport can be imported
+func (validator *VirtualMachineImportValidator) Validate(vm *ovirtsdk.Vm) ([]validators.ValidationFailure, bool) {
+	failures := validator.Validator.ValidateVM(vm)
+	if nics, ok := vm.Nics(); ok {
+		failures = append(failures, validator.Validator.ValidateNics(nics.Slice())...)
+	}
+	if das, ok := vm.DiskAttachments(); ok {
+		failures = append(failures, validator.Validator.ValidateDiskAttachments(das.Slice())...)
+	}
+
+	return processFailures(failures)
+}
+
+func processFailures(failures []validators.ValidationFailure) ([]validators.ValidationFailure, bool) {
+	valid := true
+	var propagatedFailures []validators.ValidationFailure
+	for _, failure := range failures {
+		switch checkToAction[failure.ID] {
+		case log:
+			klog.Infof("Validation information: %v", failure)
+		case warn:
+			propagatedFailures = append(propagatedFailures, failure)
+		case block:
+			valid = false
```

Comment:
What do you think that if we see any block action we return new error and stop processing?

Reply:
I wouldn't want to use error for reporting that. I see `valid==false` result as something that is an expected, correct behaviour of the application and not something abnormal that should be an error.

Reply:
Why having an error is so bad? Once we know we need to block we do not need to do anything else.

Reply:
Do you mean - breaking the loop? In that case we would have to return full slice of failures to the caller, which would have decide again which ones need to be put in the CR status.

Reply:
yes, I meant breaking the loop (stop processing). Having meaningful error would be enough. We could update vm import object with this message and mark it as failed. If you see any better way to propagate this information feel free to implement.

Reply:
We need to communicate/preserve all of the issues blocking the import so the user can fix them all and re-submit the CR without iterating from error to error. I will update the CR in that function and won't return the failures to the caller. In that case returning `error` IMO would be acceptable instead of a `bool`.